flair/training_utils.py

```diff
@@ -1,22 +1,27 @@
 import logging
+import pathlib
 import random
 from collections import defaultdict
 from enum import Enum
 from functools import reduce
 from math import inf
 from pathlib import Path
-from typing import Literal, Optional, Union
+from typing import Literal, NamedTuple, Optional, Union
 
+from numpy import ndarray
 from scipy.stats import pearsonr, spearmanr
+from scipy.stats._stats_py import PearsonRResult, SignificanceResult
```

Comment:
I appreciate that you are improving the typing information, but in this case it is adding an import to a private module.
In that case, I would prefer to not have that function typed.

Reply:
No problem, removed